plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/AbstractTestOracleTypeMapping.java

```diff
@@ -788,56 +787,72 @@ public void testTimestamp(boolean insertWithTrino, ZoneId sessionZone)
+                .addRoundTrip("timestamp with time zone", "TIMESTAMP '2019-03-18 10:01:17.987 +02:17'", TIMESTAMP_TZ_MILLIS, "TIMESTAMP '2019-03-18 10:01:17.987 +02:17'")
+                .addRoundTrip("timestamp with time zone", "TIMESTAMP '2019-03-18 10:01:17.987 -07:31'", TIMESTAMP_TZ_MILLIS, "TIMESTAMP '2019-03-18 10:01:17.987 -07:31'")
+                .addRoundTrip("timestamp with time zone", "TIMESTAMP '2018-10-28 01:33:17.456 Z'", TIMESTAMP_TZ_MILLIS, "TIMESTAMP '2018-10-28 01:33:17.456 Z'")
+                .addRoundTrip("timestamp with time zone", "TIMESTAMP '2018-10-28 01:33:17.456 America/Bahia_Banderas'", TIMESTAMP_TZ_MILLIS, "TIMESTAMP '2018-10-28 01:33:17.456 America/Bahia_Banderas'")
+                .addRoundTrip("timestamp with time zone", "TIMESTAMP '2018-10-28 01:33:17.456 Asia/Kathmandu'", TIMESTAMP_TZ_MILLIS, "TIMESTAMP '2018-10-28 01:33:17.456 Asia/Kathmandu'")
+                .addRoundTrip("timestamp with time zone", "TIMESTAMP '2018-10-28 03:33:33.333 Z'", TIMESTAMP_TZ_MILLIS, "TIMESTAMP '2018-10-28 03:33:33.333 Z'")
+                .addRoundTrip("timestamp with time zone", "TIMESTAMP '2018-10-28 03:33:33.333 Europe/Vilnius'", TIMESTAMP_TZ_MILLIS, "TIMESTAMP '2018-10-28 03:33:33.333 Europe/Vilnius'")
+                .addRoundTrip("timestamp with time zone", "TIMESTAMP '2018-10-28 03:33:33.333 Asia/Kathmandu'", TIMESTAMP_TZ_MILLIS, "TIMESTAMP '2018-10-28 03:33:33.333 Asia/Kathmandu'")
+                .addRoundTrip("timestamp with time zone", "TIMESTAMP '1970-01-01 00:13:42.000 Z'", TIMESTAMP_TZ_MILLIS, "TIMESTAMP '1970-01-01 00:13:42.000 Z'")
+                .addRoundTrip("timestamp with time zone", "TIMESTAMP '1970-01-01 00:13:42.000 Asia/Kathmandu'", TIMESTAMP_TZ_MILLIS, "TIMESTAMP '1970-01-01 00:13:42.000 Asia/Kathmandu'")
+                .addRoundTrip("timestamp with time zone", "TIMESTAMP '2018-04-01 02:13:55.123 Z'", TIMESTAMP_TZ_MILLIS, "TIMESTAMP '2018-04-01 02:13:55.123 Z'")
+                .addRoundTrip("timestamp with time zone", "TIMESTAMP '2018-04-01 02:13:55.123 Asia/Kathmandu'", TIMESTAMP_TZ_MILLIS, "TIMESTAMP '2018-04-01 02:13:55.123 Asia/Kathmandu'")
+                .addRoundTrip("timestamp with time zone", "TIMESTAMP '2018-03-25 03:17:17.000 Asia/Kathmandu'", TIMESTAMP_TZ_MILLIS, "TIMESTAMP '2018-03-25 03:17:17.000 Asia/Kathmandu'")
+                .addRoundTrip("timestamp with time zone", "TIMESTAMP '1986-01-01 00:13:07.000 Europe/Vilnius'", TIMESTAMP_TZ_MILLIS, "TIMESTAMP '1986-01-01 00:13:07.000 Europe/Vilnius'")
+                .execute(getQueryRunner(), trinoCreateAsSelect("timestamp_tz"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("timestamp_tz"));
     }
 
-    @DataProvider
-    public Object[][] testTimestampWithTimeZoneDataProvider()
+    @Test
+    public void testTimestampWithTimeZoneFromOracle()
     {
-        return new Object[][] {
-                {true},
-                {false},
-        };
+        DataTypeTest.create()
```

Comment:
Let's create an issue about the DST handling issue and add a TODO here.

Reply:
Thanks for your comment. TODO comment was added in `Convert oracle timestamp with time zone test to SqlDataTypeTest` ea41f56.

Reply:
how did the problem manifest when converting from `DataTypeTest` to `SqlDataTypeTest`?
is it because `SqlDataTypeTest` does more checks and `DataTypeTest.create(true)` would reveal the problem too?

Reply:
> how did the problem manifest when converting from DataTypeTest to SqlDataTypeTest?

As mentioned in pr desc
> there are two test cases in testTimestampWithTimeZoneFromOracle() still use DataTypeTest.
'2018-10-28 01:33:17.456 America/Bahia_Banderas'
'2018-10-28 03:33:33.333000000', 'Europe/Vilnius'
Both of them are in DST which make SqlDataTypeTest fail.
For example, 2018-10-28 01:33:17.456 America/Bahia_Banderas is stored in oracle, when trino get it back, the value is 1540711997456 in millisecond, while in trino sql values ('2018-10-28 01:33:17.456 America/Bahia_Banderas') is 1540708397456 in millisecond.
In SqlDataTypeTest, it use IS NOT DISTINCT FROM in test, since the millisecond between trino and oracle are totally different, test will fail.

It seems oracle stores different value to trino in TimestampWithTimeZone datatype.

> is it because SqlDataTypeTest does more checks and DataTypeTest.create(true) would reveal the problem too?

Indeed, DataTypeTest compare timestamp with tz result in string format, while SqlDataTypeTest use `IS NOT DISTINCT FROM`, it will compare the exact long value.

Reply:
> The values ​​stored by trino and oracle are different in Timestamp with tz type when it come to DST.

i think we do not check the value stored in oracle. maybe the value stored in oracle is simply different from the value we were supposedly inserting?

Reply:
>  maybe the value stored in oracle is simply different from the value we were supposedly inserting?

If yes, is this a correct behavior ? Sorry I'm not very familiar with timestamp in tz in DST.